code/src/components/TaskList.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import tasks from 'reducer/tasks';
+import styled from 'styled-components'
+
+const TaskList = () => {
+  const taskList = useSelector((store) => store.tasks.items)
+
+  const dispatch = useDispatch();
+
+  const onIsCompleteToggle = (id) => {
+    dispatch(tasks.actions.toggleItem(id))
+  }
+
+  const onRemoveButtonClick = (taskIndex) => {
+    dispatch(tasks.actions.removeTask(taskIndex))
+  }
+  return (
+    <section>
+      {taskList.map((singleTask) => {
+        return (
+          <StyledTask key={singleTask.id}>
+            <Checkbox
+              type="checkbox"
+              name="tasks"
+              checked={singleTask.complete}
+              onChange={() => onIsCompleteToggle(singleTask.id)}
+              id={singleTask.id} />
+            <div>{singleTask.text}
+              <StyledTimeStamp>
+                {singleTask.timeStamp.toLowerCase()}
+              </StyledTimeStamp>
+            </div>
+            <StyledDeleteButton type="button" onClick={onRemoveButtonClick}>
```

Comment:
I think here you don't pass an ID to the delete-function, maybe that's the problem?